spec/acceptance/self_service_spec.rb

```diff
@@ -262,6 +263,43 @@ class {'self_service':
         expect(result.stdout).to match(%r{false})
         run_shell('touch -d "1 day ago"  /opt/puppetlabs/server/pe_build')
       end
+      it 'if S0035 conditions for false are met' do
+        # create /opt/puppetlabs/puppet/modules/dummy module with invalid dependency metadata.json
+        metadata_json = <<~'METADATA'
+            {
+                "name": "user-dummy",
+                "version": "0.1.0",
+                "author": "user",
+                "summary": "dummy module with invalid dependency",
+                "license": "Apache-2.0",
+                "source": ".",
+                "dependencies": [
+                  {
+                    "name": "puppetlabs/stdlib",
+                    "version_requirement": "<= 0.1.0"
+                  }
+                ]
+            }
+            METADATA
+        
+        # create temporary dummy module directory and ensure it gets deleted
+        dir = File.join('/', 'opt', 'puppetlabs', 'puppet', 'modules', 'dummy')
+        begin
+          FileUtils.mkdir_p dir
+          File.open(File.join(dir, 'metadata.json'), 'w') { |file| file.write(metadata_json) }
+
+          # verify that puppet module list returns a warning, i.e., S0035 => false
+          result = run_shell('facter -p self_service.S0035')
+          expect(result.stdout).to match(%r{false})
+        ensure
+          # delete /opt/puppetlabs/puppet/modules/dummy module
+          FileUtils.rm_rf(dir)
+        end
```

Comment:
@MartyEwings:  I've added this begin, ensure, end to hopefully ensure the test directory gets deleted.  On some of the OS's the temporary directory wasn't getting deleted and the final check for S0005 => true was failing.